components/binding-usage-controller/internal/controller/controller.go

```diff
@@ -207,12 +207,21 @@ func (c *ServiceBindingUsageController) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(key)
 
-	err := c.syncServiceBindingUsage(key.(string))
+	namespace, name, err := cache.SplitMetaNamespaceKey(key.(string))
+	if err != nil {
+		c.log.Errorf("Error processing %q (splitting meta namespace key failed): %v", key, err)
+		c.queue.Forget(key)
+		return true
+	}
+
+	retry := c.maxRetires
```

Comment:
`retries := c.queue.NumRequeues(key)`,  as discussed.
Please use `retries` instead of each occurrence `c.queue.NumRequeues(key)` occurence.